### ISSUE TITLE
XAML TextBlock with IsTextSelectionEnabled is slow when selecting text in long string of Emojis

### ISSUE THREAD

**Issue by lyahdav** [microsoft/microsoft-ui-xaml#7122]
### Describe the bug

If you create a XAML TextBlock with IsTextSelectionEnabled and a long string of Emojis, performance is very slow when changing the selected text via mouse drag.

### Steps to reproduce the bug

1. Open [this solution](https://github.com/lyahdav/XamlPlayground/blob/slow-text-block-emoji-selection/XamlPlayground.sln).
2. Run the app.
3. Mouse drag to change text selection in the Emoji string and notice very slow performance.

### Expected behavior

Dragging over long string of emojis should have similar performance to non-emoji text.

### Screenshots

Here's a video of the app from the steps above. When the mouse is down a circle is rendered. You can see that dragging over the emojis is very slow to update the selected state, but with plain text it's much faster.

https://user-images.githubusercontent.com/359157/169176182-902a7948-4eab-463c-b00a-c87c1d334fa0.mov

### Windows app type

- [X] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

Windows 10 (21H2): Build 19044

### Additional context

The code to reproduce the issue is [here](https://github.com/lyahdav/XamlPlayground/blob/slow-text-block-emoji-selection/XamlIslandsApp/Shared.cpp#L10).
Note that I only tested in a UWP and XAML Islands app, not WinUI 3.

**Comment by lyahdav** [comment]
Likely still relevant

**Comment by JJBrychell** [comment]
Placing this much text into a text block is not a high priority scenario for TextBlocks which are designed for labels and descriptive text.  The repro uses around 4000 characters, and I have confirmed that up to around 1500 characters things seem to perform fine. There MAY be things that could be done to improve the performance here, but without a compelling set of scenarios, it is unlikely that we would invest our time here.

Closing this issue as not planned.
